homeassistant/components/izone/discovery.py

```diff
@@ -0,0 +1,133 @@
+"""Internal discovery service for  iZone AC."""
+
+import logging
+from asyncio import Event
+from typing import Dict
+
+import pizone
+
+from homeassistant.const import CONF_EXCLUDE, EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .climate import ControllerDevice
+from .constants import DATA_ADD_ENTRIES, DATA_CONFIG, DATA_DISCOVERY_SERVICE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_start_discovery_service(hass: HomeAssistantType):
+    """Set up the pizone internal discovery."""
+    class DiscoveryService(pizone.Listener):
+        """Discovery data and interfacing with pizone library."""
+
+        def __init__(self):
+            self.controllers = {}  # type: Dict[str, pizone.Controller]
+            self.controller_ready = Event()
+            self.components = {}  # type:  Dict[pizone.Controller, 'IZoneController']
+
+            self.pi_disco = None
+            self.stop_listener = None
+
+        # Listener interface
+        def controller_disconnected(self, ctrl: pizone.Controller,
+                                    ex: Exception) -> None:
+            """Disconnected from contrller."""
+            component = self.components.get(ctrl)
+            if not component:
+                return
+            component.set_available(False)
+
+        def controller_reconnected(self, ctrl: pizone.Controller) -> None:
+            """Reconnected to controller."""
+            component = self.components.get(ctrl)
+            if not component:
+                return
+            component.set_available(True)
+
+        async def _controller_discovered(self, ctrl: pizone.Controller):
+            _LOGGER.debug("Controller discovered uid=%s", ctrl.device_uid)
+
+            conf: ConfigType = hass.data.get(DATA_CONFIG)
```

Comment:
That doesn't look like py3.5 compatible syntax :P

Reply:
Yeah. Weeding them out. I need to install 3.5 on my dev envt.